### DISCUSSION TITLE
Why is the inverse transformation of the target not applied? I think it's a bug

### DISCUSSION THREAD

**Discussion by acartro** [pycaret/pycaret#2706] [Q&A]
Good morning,

I am doing a time series experiment with the new 3.0 library. I have found that when I set the setup to do a logarithmic transformation of the target (transform_target='log') when making the forecasts, it applies the inverse transformation. 

### Whit target transformation
```
from pycaret.time_series import *
from pycaret.datasets import get_data
y = get_data("airline")

#### Setup experiment ----
exp = TSForecastingExperiment()
exp.setup(data=y, fh=12, session_id=42, transform_target='log')

##### List the available models ----
exp.models()

#### Create a model from available models ----
model = exp.create_model("arima")

### plot model
exp.plot_model(model )
```
![image](https://user-images.githubusercontent.com/11137446/177162592-c0e9c57e-0bf9-43ba-a8c5-909c9caa3226.png)

### Whit target transformation and impute missings

But when I set the setup to do an imputation of the missings and then the logarithmic transformation (transform_target='log', numeric_imputation_target='drift'), when it makes the forecast, it doesn't apply the inverse transformation.

```
#### Setup experiment ----
exp = TSForecastingExperiment()
exp.setup(data=y, fh=12, session_id=42, transform_target='log',
          numeric_imputation_target='drift')

##### List the available models ----
exp.models()

#### Create a model from available models ----
model = exp.create_model("arima")

### plot model
exp.plot_model(model )
```
![image](https://user-images.githubusercontent.com/11137446/177163187-fcb08417-ad58-4eaf-82b2-37bbc1346c21.png)

Does anyone know why this is happening or am I just doing the forecast wrong?

**Comment by ngupta23** [comment]
What version of pycaret are you using. Can you post the output of show_versions()

**Reply by acartro** [reply]
Of course

```
import pycaret
print(pycaret.__version__)
```
![image](https://user-images.githubusercontent.com/11137446/177205619-d9d3b871-8f59-4797-bc67-218bb728469a.png)

and 

```
from pycaret import show_versions
show_versions()
```
![image](https://user-images.githubusercontent.com/11137446/177205768-072e844a-dbcc-451e-85dc-3706dc6f3582.png)

**Comment by acartro** [comment]
I think it's a bug.

It happens not only with transform_target but also with scale_target. 
If you apply one of these two parameters in the setup together with numeric_imputation_target or numeric_imputation_exogenous or both together. It does not perform the inverse transformation

Here are some examples if you want:

```
# loading dataset
import numpy as np
from pycaret.time_series import *
from pycaret.datasets import get_data
data = get_data('airline')
# setup experiment
s = setup(data, fh=3, fold=5, session_id=123, verbose=False, transform_target='box-cox',  numeric_imputation_target='ffill',
          numeric_imputation_exogenous='ffill'
          )
# create model
model = create_model("arima")
# predict model
predict_model(model)
```
![image](https://user-images.githubusercontent.com/11137446/178811598-45ab59b4-af21-444f-bfac-ec66fe0fc06c.png)

```
# loading dataset
import numpy as np
from pycaret.time_series import *
from pycaret.datasets import get_data
data = get_data('airline')
# setup experiment
s = setup(data, fh=3, fold=5, session_id=123, verbose=False, scale_target='zscore',  numeric_imputation_target='ffill',
          numeric_imputation_exogenous='ffill'
          )

# create model
model = create_model("arima")
# predict model
predict_model(model)
```

![image](https://user-images.githubusercontent.com/11137446/178812003-de08ad13-71ee-4a24-9d76-e0079306f529.png)

**Comment by ngupta23** [comment]
TRacked here: https://github.com/pycaret/pycaret/issues/2771
